tests/ui/parser/issues/issue-104088.rs

```diff
@@ -8,10 +8,12 @@ fn 1234test() {
 
     let 23name = 123;
     //~^ ERROR expected identifier, found `23name`
-
+}
```

Comment:
can we move this test to `recovery/recover-number-ident` or something like that? it's not a regression test for a weird case so i think it's pretty misplaced here